// Copyright AGNTCY Contributors (https://github.com/agntcy)
// SPDX-License-Identifier: Apache-2.0

syntax = "proto3";

package routing.v1;

// A query to match the record against during discovery.
// For example:
//  { type: RECORD_QUERY_TYPE_SKILL, value: "Natural Language Processing" }
//  { type: RECORD_QUERY_TYPE_LOCATOR, value: "helm-chart" }
message RecordQuery {
  // The type of the query to match against.
  RecordQueryType type = 1;

  // The query value to match against.
  string value = 2;
}

// Defines a list of supported record query types.
enum RecordQueryType {
  // Unspecified query type.
  RECORD_QUERY_TYPE_UNSPECIFIED = 0;

  // Query for a skill name.
  RECORD_QUERY_TYPE_SKILL = 1;

  // Query for a locator type.
  RECORD_QUERY_TYPE_LOCATOR = 2;
}
